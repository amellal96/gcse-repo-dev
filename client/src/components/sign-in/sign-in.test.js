import React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk'
import {Provider} from 'react-redux';
import { mount } from 'enzyme';

jest.mock('../../actions/questions');
jest.mock('../../actions/user');

import SignIn from './sign-in.component';

const mockStore = configureStore([thunk]);

describe('BrowseContainer component', () => {
    let store;
    let wrapper;
    let mockLogin;
    let mockData;

    beforeEach(() => {
        store = mockStore({user: {
            user: {
                isAuthenticated: false
            }
        }});

        store.dispatch = jest.fn();
        mockLogin = jest.fn();
        mockData = {};

        const mockProps = {
            store,
            login: mockLogin
        }

        wrapper = mount(<Provider store={store}>
            <SignIn {...mockProps} />
        </Provider>);
    }); 

    it('should match the snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot(); 
    });

    it('should render sign in form', () => {
        expect(wrapper.find('.sign-in-form')).toHaveLength(1);
    });
});