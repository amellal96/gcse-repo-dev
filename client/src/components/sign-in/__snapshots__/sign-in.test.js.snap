// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrowseContainer component should match the snapshot 1`] = `"<div class=\\"sign-in\\"><h2 class=\\"title\\">I have an account</h2><form class=\\"sign-in-form\\"><div class=\\"group\\"><label class=\\" form-input-label\\">Email</label><input class=\\"form-control\\" type=\\"email\\" name=\\"email\\" required=\\"\\" value=\\"\\"></div><div class=\\"group\\"><label class=\\" form-input-label\\">Password</label><input class=\\"form-control\\" type=\\"password\\" name=\\"password\\" required=\\"\\" value=\\"\\"></div><button class=\\"  custom-button\\" type=\\"submit\\">SIGN IN</button></form></div>"`;
